📖 Improve Go bump instructions (#12246)
kubernetes-sigs/cluster-api: docs/release/role-handbooks/release-lead/README.md

> Note: If the Go minor version of one of our supported branches goes out of supported, we should consider bumping

Reviewer: ```suggestion
Note: If the Go minor version of one of our supported branches goes out of support, we should consider bumping
```
If you don't mind, could you kindly fix this nit

Author: Sure, no problem

to a newer Go minor version according to our [backport policy](./../../../../CONTRIBUTING.md#backporting-a-patch).
Usually this is done after k/k bumps the Go version in their release branches. For example:

> * Cluster API v1.10 uses Kubernetes v1.32: https://github.com/kubernetes-sigs/cluster-api/blob/v1.10.0/go.mod

Reviewer: May be
```suggestion
* Cluster API v1.10 has Kubernetes v1.32 as a dependency, which originally was using Go 1.22: https://github.com/kubernetes-sigs/cluster-api/blob/v1.10.0/go.mod
```

Author: Added the first half the suggestion. Not sure if the latter is helpful. Similar explanation as above.

The whole text here starts with "If the Go minor version of one of our supported branches goes out of support, we should consider bumping"

And this listing here is to explain how to lookup the right Go version we should use (I'll change the text above the listing)

> * Kubernetes v1.32 uses Go 1.23.8: https://github.com/kubernetes/test-infra/blob/1a5662a/images/kubekins-e2e-v2/variants.yaml#L19

Reviewer: May be
```suggestion
* Kubernetes v1.32 branch bumps to Go 1.23.8 for any reason (and a coresponding kubekins image is available in https://github.com/kubernetes/test-infra/blob/1a5662a/images/kubekins-e2e-v2/variants.yaml#L19)
```
